native/cocos/engine/Engine.cpp

```diff
@@ -97,6 +97,8 @@ namespace cc {
 Engine::Engine() {
     _scheduler = std::make_shared<Scheduler>();
     _fs = createFileUtils();
+    // May create gfx device in render subsystem in future.
+    _gfxDevice = gfx::DeviceManager::create();
```

Comment:
If we create gfx device before script engine, user might lost device configuration ability in js. 
However, as in comment, we may create gfx device in render subsystem in the future.

Reply:
Yep, i think we need a render subsystem in future, and gfx device will be created there.